### PR TITLE
Created SeoAwareTrait

### PR DESCRIPTION
**Pull request by core23**
I am targeting this branch, because this is a new feature.

## Changelog

```markdown
### Added
- Added `SeoAwareTrait`
```

## Subject

This trait can be used the same way as the `LoggerAwareTrait`, `ContainerAwareTrait`, ...

### CODE REVIEW

File: src/Seo/SeoAwareTrait.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\SeoBundle\Seo;
+
+trait SeoAwareTrait
+{
+    /**
+     * @var SeoPageInterface|null
```
Comment (kunicmarko20) [inline src/Seo/SeoAwareTrait.php]: What about adding `@required` it will try to autowire it.

Comment (core23) [reply src/Seo/SeoAwareTrait.php]: Didn't find any examples in the other traits.
Should you use autowiring when there is a setter method?

Comment (kunicmarko20) [reply src/Seo/SeoAwareTrait.php]: why not? You can still set it yourself or let it be autowired. A better question is, should we autowire it at all, I think it is not a best practice for 3rd party bundles.

**Comment by OskarStark** [comment]
Do we need to tweak the docs or add a cookbook article about it?

**Comment by kunicmarko20** [comment]
Yes, people need to know about it so they can use it.

File: src/Seo/SeoAwareTrait.php (hunk 1 above)
Comment (core23) [reply src/Seo/SeoAwareTrait.php]: Fixed id

**Comment by greg0ire** [comment]
Thanks @core23 !

**Pull request merged**